libraries/rush-lib/src/logic/selectors/PathProjectSelectorParser.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as nodePath from 'node:path';
+
+import { AlreadyReportedError } from '@rushstack/node-core-library';
+import type { LookupByPath } from '@rushstack/lookup-by-path';
+
+import type { RushConfiguration } from '../../api/RushConfiguration';
+import type { RushConfigurationProject } from '../../api/RushConfigurationProject';
+import type { IEvaluateSelectorOptions, ISelectorParser } from './ISelectorParser';
+import { RushConstants } from '../RushConstants';
+
+export class PathProjectSelectorParser implements ISelectorParser<RushConfigurationProject> {
+  private readonly _rushConfiguration: RushConfiguration;
+  private readonly _workingDirectory: string;
+
+  public constructor(rushConfiguration: RushConfiguration, workingDirectory?: string) {
```

Comment:
Seems like `workingDirectory` should always be defined, as this is an internal API and the Rush CLI should really only have one place where `process.cwd()` is touched.